src/main/java/feiner/rijks/RijksSearchFrame.java

```diff
@@ -0,0 +1,183 @@
+                    .subscribe(
+                            (response) -> handleResponse(response),
+                            Throwable::printStackTrace);
+        }
+    }
+
+    private void onPrevious() {
+        pageNumber--;
+        if (pageNumber < 0) {
+            pageNumber = 0;
+        }
+        String query = searchBar.getText();
+        if (!query.isEmpty()) {
+            Disposable disposable = service.querySearch(keyString, pageNumber, query)
+                    // tells Rx to request the data on a background Thread
+                    .subscribeOn(Schedulers.io())
+                    // tells Rx to handle the response on Swing's main Thread
+                    .observeOn(SwingSchedulers.edt())
+                    //.observeOn(AndroidSchedulers.mainThread()) // Instead use this on Android only
+                    .subscribe(
+                            (response) -> handleResponse(response),
+                            Throwable::printStackTrace);
+        } else {
+            Disposable disposable = service.pageSearch(keyString, pageNumber)
+                    // tells Rx to request the data on a background Thread
+                    .subscribeOn(Schedulers.io())
+                    // tells Rx to handle the response on Swing's main Thread
+                    .observeOn(SwingSchedulers.edt())
+                    //.observeOn(AndroidSchedulers.mainThread()) // Instead use this on Android only
+                    .subscribe(
+                            (response) -> handleResponse(response),
+                            Throwable::printStackTrace);
+        }
+    }
+
+    private void onSearch() {
+        pageNumber = 0;
+        String query = searchBar.getText();
+            Disposable disposable = service.querySearch(keyString, pageNumber, query)
+                    // tells Rx to request the data on a background Thread
+                    .subscribeOn(Schedulers.io())
+                    // tells Rx to handle the response on Swing's main Thread
+                    .observeOn(SwingSchedulers.edt())
+                    //.observeOn(AndroidSchedulers.mainThread()) // Instead use this on Android only
+                    .subscribe(
+                            (response) -> handleResponse(response),
+                            Throwable::printStackTrace);
+        }
+
+    private void handleResponse(ArtObjectCollection response) {
+        imagePanel.removeAll();
+        Executors.newSingleThreadExecutor().execute(() -> {
+            try {
+                for (ArtObject artObject : response.artObjects) {
+                    JLabel label = createImageLabel(artObject);
+                    SwingUtilities.invokeLater(() -> imagePanel.add(label));
+                }
+                SwingUtilities.invokeLater(() -> {
+                    imagePanel.revalidate();
+                    imagePanel.repaint();
+                });
```

Comment:
I'm not sure this is necessary

Reply:
It says Deprecated Gradle features were used in this build, making it
incompatible with Gradle 9.0.

Reply:
That's nothing to worry about.